### DISCUSSION TITLE
Using wamrc for target aarch64 throws an error

### DISCUSSION THREAD

**Discussion by schorg** [bytecodealliance/wasm-micro-runtime#1944] [Q&A] [answered]
I have a C application, compiled with wasi-sdk to wasm.
When I try to aot compile for target architecture aarch64 I get the following error:

```
➜ wamrc --target=aarch64 -o demo_io.aot demo_io.wasm
Create AoT compiler with:
  target:        aarch64v8
  target cpu:    
  cpu features:  
  opt level:     3
  size level:    1
  output format: AoT file
LLVM ERROR: Only small, tiny and large code models are allowed on AArch64
[1]    69636 abort      wamrc --target=aarch64 -o demo_io.aot demo_io.wasm
```
Any idea, how to fix this? Thanks for any hints.

**Comment by loganek** [comment, marked as answer]
yes, medium code model is not supported on AArch64 (and it's apparently the default one for MacOS and Windows). You can try experimenting with different models by passing `--size-level=` parameters to `wamrc`

**Reply by schorg** [reply]
I'll try again to aot-compile on Linux, if this helps.

**Reply by loganek** [reply]
you can also try it on macos/windows by setting `--size-level` parameter (e.g. `--size-level=3` will set the code model to Small, which is supported on AArch64 IIRC)

**Reply by schorg** [reply]
I am working on a Mac M1, but I have also a VM with Ubuntu running. I compiled `wamrc` on the VM and it worked. 
I will later try your proposal on MacOS. But for now my problem is solved.
Thank you very much.

**Reply by eloparco** [reply]
If you change the size level you'll be able to compile on your Mac M1, but then you'll have problems running it https://github.com/bytecodealliance/wasm-micro-runtime/issues/1365. I usually use my Linux machine for AOT.
